core/src/test/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolverTests.java

```diff
@@ -970,4 +969,17 @@ public void testIndexAliases() {
         Arrays.sort(strings);
         assertArrayEquals(new String[] {"test-alias-0", "test-alias-1", "test-alias-non-filtering"}, strings);
     }
+
+    public void testConcreteIndicesForDeprecatedPattern() {
```

Comment:
could you slightly expand this test by adding the different combinations that we used to have above? This is just to make sure that everything still works properly.